node/external/interface.go

```diff
@@ -1,13 +1,17 @@
 package external
 
 import (
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
 	"github.com/ElrondNetwork/elrond-go/process"
 	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
 )
 
 // SCQueryService defines how data should be get from a SC account
 type SCQueryService interface {
 	ExecuteQuery(query *process.SCQuery) (*vmcommon.VMOutput, error)
+	ComputeTransactionCost(tx *transaction.Transaction) (*big.Int, error)
```

Comment:
query service should really do only querry for smart contract. computate cost can be on top of that.

Reply:
done.